### ISSUE TITLE
Common building and deploying instructions

### ISSUE THREAD

**Issue by warpech** [Starcounter/Guidelines#17]
Right now all of our sample apps have different instructions for building and deploying to Warehouse. 

The instructions should be common. They should be in CONTRIBUTING.md.

@Mackiovello could you try to propose a PR in one of the apps that contains the good instructions? It might be based on the following examples:

https://github.com/StarcounterApps/Cashier/blob/master/CONTRIBUTING.md
https://github.com/StarcounterApps/Products/blob/develop/CONTRIBUTING.md

I think that deploying to Warehouse instructions can be replaced by just a link to https://github.com/Starcounter/RebelsLounge/blob/master/apps_ecosystem/releasing_apps_to_the_warehouse.md

**Comment by Mackiovello** [comment]
> @Mackiovello could you try to propose a PR in one of the apps that contains the good instructions?

Yea, I'll do that.

Wouldn't the instructions be identical for all the apps so that we just need to have them in one place? Copying to all the `CONTRIBUTING.md` feels like a massive violation of the DRY principle.

**Comment by warpech** [comment]
> Wouldn't the instructions be identical for all the apps so that we just need to have them in one place? Copying to all the `CONTRIBUTING.md` feels like a massive violation of the DRY principle. 

Yes, they should exactly the same, but I want them to be discoverable. For me it is fine to have just a short `CONTRIBUTING.md` file everywhere with a link to a relevant page in https://starcounter.gitbooks.io/guidelines/content/

**Comment by Mackiovello** [comment]
Is there still a need for the old warehouse? It says in the [release instructions](https://github.com/Starcounter/RebelsLounge/blob/master/apps_ecosystem/releasing_apps_to_the_warehouse.md) that: "The apps that work with Starcounter 2.3 and above, should be uploaded to the new Warehouse."

It also explains that: "for other apps [that are not on TeamCity], install Starcounter stable and test manually". When it says "test manually" does that mean just building for apps that don't have tests?

**Comment by tomalec** [comment]
> For me it is fine to have just a short CONTRIBUTING.md file everywhere with a link to a relevant page in https://starcounter.gitbooks.io/guidelines/content/

Will it then still look good once GitHub notifies with contributing? I think similar problem applies to other files as well: `.travis.yml` in Juicy, `.bowerrc` in Apps, `LICENSE`everywhere , etc. Maybe we could have gitbot for such things?

**Comment by warpech** [comment]
> Is there still a need for the old warehouse? It says in the release instructions that: "The apps that work with Starcounter 2.3 and above, should be uploaded to the new Warehouse."

I think there is no need for deploying sample apps to the old Warehouse anymore since we changed the default download to Starcounter 2.3.x.

> When it says "test manually" does that mean just building for apps that don't have tests?

It means build  and run it to do a quick smoke test of it.

**Comment by warpech** [comment]
> Also, in those instructions, committing comes after packaging. Wouldn't it be better to reverse that order so that the zip file isn't committed?

It might happen that Warehouse rejects the version for some reason (wrong metadata, wrong version number, etc). So I would rather commit only after Warehouse has accepted the package. But of course the ZIP should not be committed.

**Comment by Mackiovello** [comment]
The releasing apps to warehouse page has been [updated](https://github.com/Starcounter/RebelsLounge/blob/master/apps_ecosystem/releasing_apps_to_the_warehouse.md) according to your comments

I've also added instructions on how to contribute in https://github.com/StarcounterApps/Products/pull/43/commits/6349af5df1f7917682917a350fbeb3161268f1f2

**Comment by warpech** [comment]
Great! 

I have fixed some typos in (releasing_apps_to_the_warehouse.md)[https://github.com/Starcounter/RebelsLounge/blob/master/apps_ecosystem/releasing_apps_to_the_warehouse.md].

Contributing code instructions look nice, but I think that they also belong to the Guidelines :)

**Comment by andwah** [comment]
How `Starpack` picks the version from the assembly (binary file)

File: AssemblyInfo.cs
```
...
[assembly: AssemblyVersion("1.0.0.0")]
[assembly: AssemblyFileVersion("1.0.0.0")]
[assembly: AssemblyInformationalVersion("1.0.0.0")]
...
```
Priority:
1. `AssemblyInformationalVersion`
2. `AssemblyFileVersion`
3. `AssemblyVersion`
4. 0.0.0.0 if none exists

Maybe for the docs?

**Comment by Mackiovello** [comment]
> Contributing code instructions look nice, but I think that they also belong to the Guidelines :)

There are now instructions here: https://starcounter.gitbooks.io/guidelines/content/contributing-code.html

**Comment by Mackiovello** [comment]
Common [contributing](https://starcounter.gitbooks.io/guidelines/content/contributing-code.html) and [releasing to warehouse](https://starcounter.gitbooks.io/guidelines/content/releasing-to-warehouse.html) instructions are now in the guidelines.

Building is unified by default now since the batch scripts handle building now. It just has to be included to the CONTRIBUTING files of the apps.

**Comment by warpech** [comment]
Great! @Mackiovello could you please proceed with PRs in the apps?

**Comment by Mackiovello** [comment]
The most important apps are now covered by the new guidelines.

Remaining are the following:

* [Calendar](https://github.com/StarcounterApps/Calendar)
* [WebShop](https://github.com/StarcounterApps/WebShop)
* [GoogleSignIn](https://github.com/StarcounterApps/GoogleSignIn)
* [UsageTracker](https://github.com/StarcounterApps/UsageTracker)
* [UserProfile](https://github.com/StarcounterApps/UserProfile)
* [BrainTreeWrapper](https://github.com/StarcounterApps/BraintreeWrapper)
* [LogStreamer](https://github.com/StarcounterApps/LogStreamer)
* [Simplified](https://github.com/StarcounterApps/Simplified)

It's not strictly obvious how the contributing instructions should be for those apps. I'll look closer at each of the cases and create a PR. 

Calendar does not work without PropertyMetadata
WebShop is dependent on BrainTreeWrapper https://github.com/StarcounterApps/WebShop/issues/29
GoogleSignIn has no obvious entry point https://github.com/StarcounterApps/GoogleSignIn/issues/16
UsageTracker suffers from https://github.com/Starcounter/Home/issues/89
UserProfile suffers from https://github.com/Starcounter/Home/issues/89
BrainTreeWrapper does not work with batch scripts (probably my fault)
LogStreamer doesn't build
Simplified is just different

**Comment by warpech** [comment]
Thanks for the heads up!

> Calendar does not work without PropertyMetadata

Just add a paragraph about that in `CONTRIBUTING.md`.

> WebShop is dependent on BrainTreeWrapper https://github.com/StarcounterApps/WebShop/issues/29

Maybe just add an info about it to `CONTRIBUTING.md` for the time being?

> GoogleSignIn has no obvious entry point https://github.com/StarcounterApps/GoogleSignIn/issues/16
> UsageTracker suffers from https://github.com/Starcounter/Home/issues/89
> UserProfile suffers from https://github.com/Starcounter/Home/issues/89

This means that GoogleSignIn, UsageTracker, UserProfile will build and run just fine. There is just nothing displayed at their entry point, right? That should be another issue.

> BrainTreeWrapper does not work with batch scripts (probably my fault)
> LogStreamer doesn't build

Pls check and make an issue if needed

> Simplified is just different

Yeah, we should not consider Simplified in this issue at all.

**Comment by Mackiovello** [comment]
All the apps are updated except three. I have created individual issues for them. Closing this issue as done.
